app/models/expression_file_info.rb

```diff
@@ -0,0 +1,26 @@
+class ExpressionFileInfo
```

Comment:
I'm personally not really sold on this approach.  If our end goal is a way to attach more data to a study file in an amorphous manner, but in the context of "we're going to want to search on this data later to find files", I've generally been advised by people that know Mongo that embedding documents is not the way to go.  Indexing is painful and expensive, as it needs to be defined in the parent model, so you end up with lots of large, sparse indexes because you're accounting for the whole nested structure of the child document.  Furthermore, if we extend this analogy to its logical end, we'll end up with multiple subclasses for `StudyFile` that will compound this issue further, and make querying on these attributes (which in my mind is the main use case) a pain.

The one benefit I do see to this implementation is that the subclassing makes for very clean and simple validations.  While nice, we already have compound/dependent validations in many of our models, and that complexity hasn't been an issue for us so far.  And the drawbacks to indexing/querying I feel outweigh any gains.

Reply:
Sounds like we're actually on the same page for the most part.  We agree that StudyFileTerm is the optimal structure for search, and that the structure in this PR (probably minus the `embed`) is the best structure from a non-search application perspective. 

Where I think I might not have communicated effectively is that StudyFileTerm was never intended to be a primary source of application data -- it was designed to be a supplementary search index.  Sort of like how BQ was not designed to be the final source of truth for cell metadata -- the raw files are -- BQ just makes them searchable in reasonable time.  My goal for this PR is that we continue to use the existing, solid ways we have of storing file properties as model properties, rather than putting them in a generic key-value mapping.  Then in Q1, when it comes time to implement search, if needed, we can add StudyFileTerm as a search-cache layer on top (or elasticSearch if it ends up being that we're also responsible for indexing top consortia data repo data).  So, essentially, in Q1 we just add an extra hook on study/file save that copies over any properties we're interested in searching on to the StudyFileTerm table.   

There are two things I want to avoid:  (1) implementing StudyFileTerm only to find out in Q1 that we need to implement elasticSearch anyway to support cell-level top consortia search. (2) start getting in a habit of "if it needs to be searchable, it has to go in a key-value pile (and nowhere else)".   SCP has done a fantastic job of keeping its data well-structured, even while using tools (ruby, mongo) that require a lot of discipline to avoid footgunning.  And so I don't want to start going down the key-value rabbit hole just yet... 

So I propose we model the data as in this PR (minus the embed), and then come time to implement search, decide between (1) searching as-is (2) implementing studyFileTerm as a layer on top of this (3) implementing StudyFileTerm as a replacement for this and other places where we have searchable model properties, and (4) implementing ElasticSearch

Reply:
Insightful question-and-answer, especially to this reviewer who has been less deeply involved.

@devonbush, could you add a distilled version of this summary as a file-overview comment atop the file?

Reply:
Hmmm... agree that the discussion would be nice to preserve.  I'm not sure the top of this file (a relatively obscure model) is the right place for a general discussion about the search architecture for SCP study file.  Instead I've summarized it and linked it from https://docs.google.com/document/d/1G69e80BsqUhlwRc5pqmc0WF-AfgE4DR2TdvH1Z5nPiw/edit

Reply:
That's reasonable!  Even a differently-focused overview comment (and that link, perhaps) atop this file would help, too.